File: Tutorial4-1DependentComponent/src/main/java/com/example/tutorial4_1dependentcomponent/model/Vehicle.java
package com.example.tutorial4_1dependentcomponent.model;

public class Vehicle {

    private Motor motor;

    public Vehicle(Motor motor){
        this.motor = motor;
    }

    public void increaseSpeed(int value){
        motor.accelerate(value);
    }

    public void stop(){
        motor.brake();
    }

    public int getSpeed(){
        return motor.getRpm();
    }
}

File: Tutorial4-1DependentComponent/src/main/java/com/example/tutorial4_1dependentcomponent/model/Vehicle.kt
package com.example.tutorial4_1dependentcomponent.model

class Vehicle(private val motor: Motor) {

    fun increaseSpeed(value: Int) {
        motor.accelerate(value)
    }

    fun stop() {
        motor.brake()
    }

    val speed: Int
        get() = motor.rpm

}